src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -730,6 +733,36 @@ SubCmdProgram::execute(const SubCmdOptions& _options) const
     throw xrt_core::error("uuid does not match BLP");
   }
 
+  // -- process "user" option ---------------------------------------
+  if(!xclbin.empty()) {
+    XBU::verbose(boost::str(boost::format("  xclbin: %s") % xclbin));
+    //only 1 card and name
+    if(deviceCollection.size() > 1)
+      throw xrt_core::error("Please specify a single device");
+    auto dev = deviceCollection.front();
+
+    std::ifstream stream(xclbin, std::ios::binary);
+    if (!stream)
+      throw xrt_core::error(boost::str(boost::format("Could not open %s for reding") % xclbin));
+
+    stream.seekg(0,stream.end);
+    ssize_t size = stream.tellg();
+    stream.seekg(0,stream.beg);
+
+    std::vector<char> raw(size);
+    stream.read(raw.data(), size);
+    std::string v(raw.data(),raw.data()+7);
+    if (v != "xclbin2")
+      throw xrt_core::error(boost::str(boost::format("Bad binary version '%s'") % v));
+    
+    auto bdf = xrt_core::query::pcie_bdf::to_string(xrt_core::device_query<xrt_core::query::pcie_bdf>(dev));
+    std::cout << "Downloading xclbin on device [" << bdf << "]..." << std::endl;
+    dev->load_xclbin(raw);
+    std::cout << boost::str(boost::format("INFO: Successfully downloaded xclbin \n\n"));
```

Comment:
Boost format knows out to write to an istream.  Therefore, there is no need to convert it to a string.

Reply:
Yes. Thanks :)